source/tests/system/nixnetsocket_driver_api_tests.cpp

```diff
@@ -495,19 +495,149 @@ TEST_F(NiXnetSocketLoopbackTests, OpenedStackThenCloseOriginal_GetStackInfoFromO
+  const auto closed_original_stack = client::ip_stack_clear(stub(), original_stack.stack_ref());
+
+  const auto stack_info = client::ip_stack_get_info(stub(), original_stack.stack_ref());
+  const auto reopened_stack_info = client::ip_stack_get_info(stub(), reopened_stack.stack_ref());
+  EXPECT_XNET_ERROR(INVALID_IP_STACK_ERROR, INVALID_IP_STACK_ERROR, INVALID_IP_STACK_MESSAGE, stack_info);
+  EXPECT_XNET_ERROR(INVALID_IP_STACK_ERROR, INVALID_IP_STACK_ERROR, INVALID_IP_STACK_MESSAGE, reopened_stack_info);
+}
+
+SockAddr lookup_ipv4_sock_addr(client::StubPtr& stub, const nidevice_grpc::Session& stack, const std::string& addr, int32_t port)
+{
+  const auto p_to_n = client::inet_p_to_n(stub, stack, 2 /* AF INET*/, addr);
+  auto sock_addr = SockAddr{};
+  sock_addr.mutable_ipv4()->mutable_addr()->CopyFrom(p_to_n.dst().ipv4());
+  sock_addr.mutable_ipv4()->set_port(port);
+  return sock_addr;
+}
+
+SockOptData create_membership_request(const SockAddr& multicast_addr, const SockAddr& interface_addr)
+{
+  auto opt_data = SockOptData{};
+  auto mreq = opt_data.mutable_data_ip_mreq();
+  mreq->mutable_imr_multiaddr()->CopyFrom(multicast_addr.ipv4().addr());
+  mreq->mutable_imr_interface()->CopyFrom(interface_addr.ipv4().addr());
+  return opt_data;
+}
+
+TEST_F(NiXnetSocketLoopbackTests, IPv4SocketsOnWiredPhysicalLoopback_SendAndReceiveData_ReceivesExpectedData)
 {
   const auto MESSAGE = "HelloWorld"s;
-  const auto rx_stack = client::ip_stack_create(stub(), "", create_simple_config("ENET1"));
+  const auto RX_ADDR = "10.0.0.1"s;
+  const auto rx_stack = client::ip_stack_create(stub(), "", create_simple_config("ENET1", RX_ADDR));
+  const auto tx_stack = client::ip_stack_create(stub(), "", create_simple_config("ENET4", "10.0.0.2"));
+  const auto wait = client::ip_stack_wait_for_interface(stub(), rx_stack.stack_ref(), "", 5000);
+  const auto rx_socket = socket(stub(), rx_stack.stack_ref());
+  auto sock_addr = lookup_ipv4_sock_addr(stub(), rx_stack.stack_ref(), RX_ADDR, 0);
+  const auto bind = client::bind(stub(), rx_socket.socket(), sock_addr);
+  const auto get_name = client::get_sock_name(stub(), rx_socket.socket());
+  const auto listen = client::listen(stub(), rx_socket.socket(), 10);
+  const auto tx_socket = socket(stub(), tx_stack.stack_ref());
+  EXPECT_SUCCESS(tx_stack);
+  EXPECT_SUCCESS(rx_stack);
+  EXPECT_SUCCESS(wait);
+  EXPECT_SUCCESS(rx_socket);
+  EXPECT_SUCCESS(bind);
+  EXPECT_SUCCESS(get_name);
+  EXPECT_SUCCESS(listen);
+  EXPECT_SUCCESS(tx_socket);
+
+  const auto connect = client::connect(stub(), tx_socket.socket(), get_name.addr());
+  const auto accept = client::accept(stub(), rx_socket.socket());
+  const auto send = client::send(stub(), tx_socket.socket(), MESSAGE, 0);
+  const auto recv = client::recv(stub(), accept.socket(), MESSAGE.size(), 0);
+
+  EXPECT_SUCCESS(connect);
+  EXPECT_SUCCESS_WITH_STATUS(send, MESSAGE.size());
```

Comment:
nit: if you could flip the macro so it is expected, actual must of missed that in a previous PR